File: 645-set-mismatch/set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int missing, repeat;
        unordered_map <int, int> mp;

        for ( int i=1 ; i <= nums.size() ; i++) mp[i]++;
        for ( int i : nums ) mp[i]--;

        for ( auto a : mp ) {
            if ( a.second == -1 ) repeat = a.first;
            if ( a.second == 1 ) missing = a.first;
        }

        return { repeat, missing };
    }
};

File: 645-set-mismatch/set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        natural_nums = {i: 1 for i in range(1, len(nums)+1)}
        error = [0, 0]

        for i in nums:
            natural_nums[i] -= 1
        
        for key, value in natural_nums.items():
            if value == -1: error[0] = key
            if value == 1: error[1] = key
        
        return error

